HIP/hip-873.md

```diff
@@ -0,0 +1,103 @@
+
+### Motivation
+Retrieving token balance information without decimal details requires users to make multiple API calls, leading to inefficiencies and increased computational load. Decimals should be supplied with the balance amount. Specifically, users must perform additional queries to obtain the decimal count for each token associated with an account. This is crucial for determining the correct token balance. This process is cumbersome, time-consuming, and can lead to increased costs and latency in applications relying on the Hedera API.
+
+### Specification
+The proposed solution involves modifying the response structure of two endpoints:
+
+**Route:**  
+`/api/v1/tokens?account.id={account id}`
+
+**Current Response Structure:**
+```json
+{
+  "tokens": [
+    {
+      "admin_key": null,
+      "symbol": "PEC_E",
+      "token_id": "0.0.732556",
+      "type": "NON_FUNGIBLE_UNIQUE"
+    }
+  ]
+}
+```
+
+**Proposed Response Structure:**
+```json
+{
+  "tokens": [
+    {
+      "admin_key": null,
+      "symbol": "PEC_E",
+      "name": "[Token Name]",
+      "token_id": "0.0.732556",
+      "type": "NON_FUNGIBLE_UNIQUE",
+      "decimals": [Decimals]
+    }
+  ]
+}
+```
+
+**Route:**  
+`/api/v1/accounts/{account id}`
+
+**Current Response Structure:**
+```json
+{
+  "tokens": [
+    {
+      "token_id": "0.0.732556",
+      "balance": 0
+    },
+    {
+      "token_id": "0.0.868062",
+      "balance": 66625030355247581
+    }
+  ]
+}
+```
+
+**Proposed Response Structure:**
+```json
+{
+  "tokens": [
+    {
+      "token_id": "0.0.732556",
+      "balance": 0,
+      "decimals": [Decimals]
+    },
+    {
+      "token_id": "0.0.868062",
+      "balance": 66625030355247581,
+      "decimals": [Decimals]
+    }
+  ]
+}
+```
+
+The addition of the `decimals` field will provide the necessary information to accurately determine token balances without requiring additional API calls.
+
+### Rationale
+The inclusion of decimal information in token balance queries significantly reduces the number of API calls needed to accurately determine account balances. Fewer API calls should reduce the overhead costs they produce. This enhancement aligns with the principles of efficiency, scalability, and user-friendliness that are central to the design of the Hedera API. By simplifying the process of balance retrieval, this proposal aims to improve the developer experience and the performance of applications built on Hedera.
+
+### Backwards Compatibility
+This change is backwards compatible as it introduces additional fields in the API response structure without altering the existing fields or the overall behavior of the endpoints. Clients not requiring decimal information can ignore the new fields, ensuring uninterrupted service for applications built on the current API version.
+
+### Security Implications
```

Comment:
We should add the license, references, how to teach this, and rejected ideas sections. We had some rejected ideas like add to balances API, keep HAPI queries, etc.

Reply:
Adding these now

Reply:
@steven-sheehy 
In the example section, should I add the new responses to the queries for the new route, or is it ok as is since it's a subset of /tokens